### PR TITLE
feat: websearch plugin

### PR DESCRIPTION
**Pull request by abenz1267**
Hi,

i've created a basic websearch plugin that allows you to perform searches straight from anyrun.

What do you think?

Rergards

### CODE REVIEW

File: plugins/websearch/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
+use abi_stable::std_types::{ROption, RString, RVec};
+use anyrun_plugin::*;
+use serde::Deserialize;
+use std::{fs, process::Command};
+use urlencoding::encode;
+
+#[derive(Deserialize)]
+struct Config {
+    prefix: String,
+    engine: String,
```
Comment (Kirottu) [inline plugins/websearch/src/lib.rs:10]: If there are a specific set of engines, like seen in the handler, you should use an enum for this.

**Review by Kirottu — COMMENTED** [review]
Had one comment, otherwise looks pretty good to me!

File: plugins/websearch/src/lib.rs (hunk 1 above)
Comment (abenz1267) [reply plugins/websearch/src/lib.rs:10]: Hm, i'm not that familiar with Rust and i struggle to define an Enum as a value for RON.

I've changed it so you can define multiple search engines, so i'd wanna do something like `Vec<Engine>` as the config struct value, but this doesnt work. No error sadly.

Comment (Kirottu) [reply plugins/websearch/src/lib.rs]: You can simply define the search engine as an enum like this:
```rs
#[derive(Deserialize)]
enum SearchEngine {
    Google,
    DuckDuckGo,
    Bing,
}```

Then you can place it in the config struct, `serde` will deal with the deserialization of it automatically.

Comment (Kirottu) [reply plugins/websearch/src/lib.rs]: Hm, I see no problem with the code and neither does my compiler. Specifying multiple search engines seems a bit weird though.

File: plugins/websearch/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    PluginInfo {
+        name: "Websearch".into(),
+        icon: "help-about".into(),
+    }
+}
+
+#[get_matches]
+fn get_matches(input: RString, config: &Config) -> RVec<Match> {
+    if !input.starts_with(&config.prefix) {
+        ()
```
Comment (Kirottu) [inline plugins/websearch/src/lib.rs:73]: Also, this here does not do a return from the function. It returns `()` from the `if` block, which basically means nothing. To early return from a function you need to use the `return` keyword.

File: plugins/websearch/src/lib.rs (hunk 1 above)
Comment (abenz1267) [reply plugins/websearch/src/lib.rs]: well, yeah...some people wanna use multiple search engines though. dont ask me why. they do.

in the end you can specify one if you wish.

File: plugins/websearch/src/lib.rs (hunk 2 above)
Comment (abenz1267) [reply plugins/websearch/src/lib.rs]: fixed

File: plugins/websearch/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    Google,
+    Ecosia,
+    Bing,
+    DuckDuckGo,
+}
+
+impl Engine {
+    fn value(&self) -> &str {
+        match *self {
+            Engine::Google => "google.com/search?q=",
```
Comment (Kirottu) [inline plugins/websearch/src/lib.rs:20]: These URLs should probably contain the `https://` prefix.

Comment (abenz1267) [reply plugins/websearch/src/lib.rs:20]: the protocol is defined globally for all engines in line 103 `xdg-open https://{}{}`

Comment (Kirottu) [reply plugins/websearch/src/lib.rs:20]: Oh nevermind me then, I'm just blind.

**Pull request merged**